std/algorithm/searching.d

```diff
@@ -2330,6 +2330,30 @@ if (isForwardRange!R1 && isForwardRange!R2
     assert(findSkip(s, "def") && s.empty);
 }
 
+struct FindSplitResult(R1, R2) if (isForwardRange!R1 &&
+                                   isForwardRange!R2)
+{
+    this(R1 pre, R1 separator, R2 post)
+    {
+        asTuple = typeof(asTuple)(pre, separator, post);
+    }
+
+    Tuple!(R1, R1, R2) asTuple;
+
+    bool opCast(T : bool)()
+    {
+        return !asTuple[1].empty;
+    }
+
+    alias asTuple this;
+}
+
+/// Instantiator for FindSplitResult
+private auto findSplitResult(R1, R2)(R1 pre, R1 separator, R2 post)
```

Comment:
Delete these - convenience functions are needed only when public and frequent. Here, there's just a couple of calls - insert them manually.